\graphicspath{{images/}}

\section{\thesection~Introduction}
\label{sec:introduction}

% Expand motivation, describe QFA, discuss alternative approaches.

% Dummy \citet{Lawless2010} citations \citep{Heydari2016} \citep{Addinall2008}.

% The bacteria \textit{Eschericia Coli} and yeast \textit{Saccharomyces
%   cerevisae} are unicellular organisms studied as a model prokaryote
% and eukaryote respectively. They grow in colonies, where cells may (be
% clones originating from a single cell or) belong to different genetic
% strains originating from different individual cells. In favourable
% conditions, growth is exponential and this makes growth rate a major
% component of fitness; faster growing strains quickly come to dominate
% the population. Growth rate is measurable and is often used to
% determine the fitness of microbial organisms. For unicellular
% organisms, growth rate is equal to cell cycle progression
% rate. Evolutionary pressure has led to rapidly dividing organisms with
% a compact genome of essential genes. These genes have been conserved
% in other species over billions of years of evolution. The eukaryote
% \textit{S. cerevisae}, is particularly useful for the study of other
% eukaryotes such as humans.


The bacteria \textit{Eschericia Coli} and yeast \textit{Saccharomyces
  cerevisae} are unicellular organisms studied as a model prokaryote
and eukaryote respectively. They grow in colonies, where cells may (be
clones originating from a single cell or) belong to different genetic
strains originating from different individual cells. In favourable
conditions, growth is exponential and this makes growth rate a major
component of fitness; faster growing strains quickly come to dominate
the population. At a certain point growth becomes limited and a
stationary phase is reached. For unicellular organisms, growth rate is
equal to cell cycle progression rate. Evolutionary pressure has led to
rapidly dividing organisms (such as \textit{E. Coli} and \textit{S.
  cerevisae}) with compact genomes of essential genes. These genes
have been conserved in other species over billions of years of
evolution. The eukaryote \textit{S. cerevisae}, is particularly useful
for the study of other eukaryotes such as humans.

The growth rate of microbial organisms is measurable and is often used
to determine fitness. In experiments, cell cultures are commonly grown
in two types of medium. In spot tests (phenotypic array), cultures are
pinned or inoculated on the surface of a solid agar containing
nutrients. In liquid culture assays, cultures are mixed in a liquid
medium containing nutrients. In both cases cultures are incubated and
growth is observed. Identical strains can grow differently between the
two mediums and disagreement in fitness estimates is currently an
issue \citet{Baryshnikova2010} (I couldn't find a paper specifically
talking about this issue but they have a correlation plot Fig2a where
correlations are worse with a liquid culture study by Jasnos and
Korona; in fact the Baryshnikova paper Fig3c seems to say that they
had strong correlation in their ``high-resolution liquid growth
profiling study''). I do not focus on this issue and exclusively study
fitness screens using solid agar.

% Where do cells grow surfaces films.

%%% Genetic interaction, SGA and QFA %%%

% Synthetic Genetic Array (SGA) and Quantitative Fitness Analysis (QFA)
% are high-throughput methods for obtaining quantitative fitness
% estimates of microbial cultures grown on solid agar
% \citep{Baryshnikova2010sga,Banks2012}.

% Fitness estimates can be used to infer genetic interaction or drug
% response and high-throughput methods allow this to be done on a
% genome-wide scale (see e.g. \citet{Costanzo2010}). In a typical
% genetic interaction study, a strain is made with a mutation in a query
% gene. Double mutants are created by introducing a second deletion in
% this strain. Typically one query gene is used per plate with
% replicates of many different deletions. By comparing the growth of
% double mutants with a control containing a neutral deletion, genetic
% interactions can be inferred. If a strain is fitter than the control
% then the deletion is said to suppress the defect of the query gene. If
% a strain is fitter than the control then the deletion is said to
% enhance the defect of the query gene. Either scenario suggests that
% the two genes interact and have a related function. Typically one
% query gene is used per plate and many replicates of each deletion are
% contained on each plate. Many plates can be analysed in
% high-throughput to explore whole genomes.


Fitness estimates can be used to infer genetic interaction or drug
response and high-throughput methods allow this to be conducted on a
genome-wide scale (see e.g. \citet{Costanzo2010,Andrew2013}). In a
typical genetic interaction screen, a strain is made with a mutation
in a query gene. Double mutants are created by introducing a second
deletion in this strain. By comparing the growth of double mutants
with a control containing a neutral deletion, genetic interactions can
be inferred. If a strain is fitter than the control then the deletion
is said to suppress the defect of the query gene. If a strain is less
fit than the control then the deletion is said to enhance the defect
of the query gene. Either scenario suggests that the two genes
interact and have a related function. Due to redundancy, single
deletions are often non-lethal. (Remove: Knock downs and conditional
mutations can also be used.) This allowed \citet{Costanzo2010} to
explore \(\sim\)75\% of the \textit{S. cerevisae} genome.


Synthetic Genetic Array (SGA) and Quantitative Fitness Analysis (QFA)
are high-throughput methods for obtaining quantitative fitness
estimates of microbial cultures grown on solid agar
\citep{Baryshnikova2010sga,Banks2012}. Typically one query gene and
replicates of deletion are contained in a rectangular array on a solid
agar plate. Many plates with different query genes and deletions can
be grown in high-throughput to explore whole genomes. I study data
from QFA which refers to quantitative estimation of fitness by
measurement and fitting of growth curves.


In a typical QFA procedure liquid cultures are incolulated onto solid
agar (containing nutrients (already mentioned above)) often with 384
cultures in 16x24 rectangular array. Initial inoculum density can be
varied to capture more or less of the growth curve. Dilute cultures
are inoculated with \(\sim\)100 starting cells
\citep{Addinall2011}. Plates are grown in incubation. Photographs of
whole plates are taken at timepoints throughout growth which typically
covers several days and captures the exponential and stationary growth
phases. Photographs are processed by Colonyzer \citep{Lawless2010} to
produce cell density estimates from the optical density of
cultures. Past analysis independently fit the logistic model, which
describes self-limiting growth, to the timecourse of each culture and
quantitative fitness estimates are defined in terms of the growth
constant, \(r\), and carrying capacity, \(K\). In contrast, SGA
typically uses an array of 1536 pinned cultures and an endpoint assay
of culture area (a single measurement) to quantify growth. The the
differential form and solution of the logistic model
\citep{Verhulst1845} (probably don't need this reference) are given in
Equations~\ref{eq:logistic_model}, where \(C\) represents cell density
and \(C_0\) is cell density at time zero.
% Logistic model equations
\begin{subequations}
  \label{eq:logistic_model}
  \begin{align}
    &\dot{C} = rC\left(1 - \frac{C}{K}\right)\\
    &C(t) = \frac{KC_{0}e^{rt}}{K + C_{0}(e^{rt}-1)}
  \end{align}
\end{subequations}
%
Fitting the logistic model to QFA data requires plate level or culture
level parameters for \(C_{0}\) and culture level parameters for \(r\)
and \(K\) making 769 or 1152 parameters per plate of 384 cultures.


The growth constant \(r\) could be used as a fitness measure. However,
\citet{Addinall2011} define a more complicated fitness measure as the
product of Maximum Doubling Rate (MDR) and Maximum Doubling Potential
(MDP) which they derive from logistic model parameters. MDR measures
the doubling rate at the beginning of the exponential growth phase,
when growth is fastest, and MDP is the number of divisions which a
culture undergoes from inoculation to the stationary phase.

The standard logistic model (\ref{eq:logistic_model}) does not provide
a good fit to growth curves in many cases. This lead to the use of the
generalised logsitic model which better fits growth curves but uses an
extra shape parameter for each culture. \(r\) of the standard and
generalised logistic model are not equivalent or comparable so
comparison relies on MDR as a fitness measure. The analysis of QFA
data using both the standard and generalised logistic models is
available through the QFA R package \citep{qfa2016}.

% Fits of the logistic model in liquid cultures

% generalised logistic model and how many parameters




QFA
collects more information about growth by taking images of plates at
points throughout the growth curve. QFA can be performed using either
the pinned cultures (used in SGA) or dilute liquid cultures
(``spots'') of lower initial cell density. Although pinned QFA allows
for more cultures per plate (1536 vs 384 in spotted), spotted QFA
allows for more accurate fitting of growth models as growth curves are
more complete (see Figure~\ref{fig:qfa_and_sga_growth_curves})
\citep{Lawless2010}. Comparison of spotted and pinned QFA cultures in
Figure~\ref{fig:qfa_and_sga_growth_curves}c shows how spotted cultures
are composed of many individual colonies which increase in size and
thickness, whereas pinned cultures are composed of a single uniform
colony which grows radially. The number of individual colonies in a
spotted culture is high enough that lag and other stochastic effects
should average out.

% Example use of QFA by Addinall telomeres.




\subsection{\thesubsection~Subsection}

\begin{Figure}
  \centering
  \includegraphics[width=\linewidth]{p15_section/p15_section}
  \captionof{figure}{\textbf{4x5 section of a QFA plate.} Taken from a
    16x24 format solid agar plate inoculated with dilute
    \textit{S. cerevisae} cultures.
  Image captured at \(\sim\)2.5d after inoculation and incubation at
  27\(^{\circ}C\).}
  \label{fig:p15_section}
\end{Figure}


\begin{Figure}
  \centering
  \includegraphics[width=\linewidth]{stripes/final/striped}
  \includegraphics[width=\linewidth]{stripes/final/filled}
  \captionof{figure}{\textbf{An experiment designed to examine
      competition.}\\
    A) QFA plate inocluated with a more concentrated
    \textit{S. Cerevisae} inoculum (no cells inocluated on alternative
    columns).\\
    B) Same as in A, but with strains of similar growth rate
    inoculated in the positions missing in A.}
  \label{fig:stripes_images}
\end{Figure}



%%% Local Variables:
%%% mode: latex
%%% TeX-master: "report"
%%% End:
