* TODO Introduction [9/13]
  - [X] Describe yeast/bacterial growth
  - [X] Model organism can tell us about other species (humans)
  - [X] Yeast and humans are eukaryotes
  - [X] Growth rate as a proxy for fitness
    - [X] Explain why this is suitible for microbial species
  - [X] Describe growth mediums (spot tests and liquid cultures)
    - [X] Mention disagreement in growth estimates between growth mediums
  - [-] Introduce SGA and QFA - (can leave contrast until I have
    explained QFA in more detail)
    - [X] Solid agar; high-troughput; quantitative; can study whole genomes
    - [X] Explain how fitness of genetic strains can tell us about
      genetic interactions and drug responses.
    - [ ] Give examples of this use and emphasise benefits to human health.
    - [ ] Say that Adinall studied genes involved in telomere
      stabilitly because of their links with cancer and aging and
      this could have impacts for humans but do not go into any more
      detail about telomeres, cancer, drugs, etc., (won't have room
      and don't really care for this study).
  - [-] Describe QFA in more detail - "I study QFA data ->"
    - [X] Describe the experiment design/procedure
      - Dilute or more concentrated inoculation
    - [X] Describe the data that is produced and processing with Colonyzer
    - [X] Describe current analysis with THE LOGISTIC MODEL and QFA R
      - [X] Write the logistic model equation
      - [X] I mention MDR so explain this as well
    - [X] Contrast QFA with SGA
    - [ ] Previous results (?) and follow up (Holstein &
      Clark) - Check which results were actually followed up.
  - [-] Explain the motivation for a nutrient diffusion model and
    current support for an effect
    - [X] At the same time, introduce P15 and the Stripes and Filled
      plates but wait until the methods to describe in these
      experiments in more detail.
    - [ ] Briefly mention alternative hypotheses (e.g. singal,
      metabolism) or wait until discussion? I will mention metabolism
      when I explain the approximations of the competition model. I
      think that I should probably mention signal and metabolism here
      if I have room.
  - [X] Describe previous attempts to correct for competition and how
    our approach differs
  - [X] Describe previous attempts to model diffusion (Diffusion
    equation model of Reo and Korolev 2014)
    - I am planning to talk about this in the discussion so could
      introduce here and also explain why we need a less detailed
      model (limits of computational power).
  - [-] Introduce the Competition model
    - [X] Mass action kinetics and network diffusion
    - [X] Reaction and rate equations
    - [X] No numerical solution
    - [X] Cells observed; nutrients hidden
    - [X] Explain the plate and culture level parameters.
    - [X] Explain that there are fewer parameters than in the logistic
      model.
    - [X] Explain what is measureable and what we must infer. kn? C_0
      often below detectable level in QFA - write here or elsewhere?
      We explain how we infer N_0 when we talk about initial guessing.
    - [X] Justify approximations
      - [X] Mass action kinetics
      - [X] Network diffusion
      - [X] Deterministic not stochastic
      - [X] All nutrients are converted to cells
	- Methods: Implies all cultures starting with the same amount of
          nutrients end with the same amount of cells. ->
          Mention metabolism
    - [ ] Reduces to the logistic equivalent model in the independent limit
      - [ ] Can be seen as a correction for nutrient diffusion
      - [ ] Conversion equations (Not planning to inculude the full
        derivation - Link to Conor's blog?)
      - [ ] With same plate level parameters b rank is equivalent to r
        rank (also MDR and MDR*MDP) and K is constant.
  - [X] Briefly explain what we plan to do with the model
    - Fit to data to try to estimate growth parameters more reliably
  - [X] Short summary
    - More reliable measurement of growth parameters ->
    - More reliable fitness estimates ->
    - Better prediction of genetic interaction or drug response

I could also say - We hope that a model could be used to
compare/improve experimental design by simulation of
e.g. randomisation / SGA v QFA and dilute vs concentrated. I could
just say this in one line but I imagine that the above will already
take a fair amount of space and we haven't found that yet. Perhaps at
the end of the discussion.

* Methods

** TODO Tools, solving, and fitting [0/6]
  - [ ] I wrote the code in python
  - [ ] Data analysed using Colonyzer (Python)
  - [ ] Solving
    - [ ] scipy.odeint -> optimized by vectorizing and using numpy
      - [ ] atol=1.49012e-8, rtol=1.49012e-8, mxhnil=0
    - [ ] libroadrunner
    - [ ] Empties
    - [ ] Two N_0 model
    - [ ]
  - [ ] Writing models using libSBML
  - [ ] SBML and reproducibility - spit out
  - [ ] Fitting the Competition Model
    - [ ] SciPy's scipy.optimize.minimize (optimization and rootfinding package)
    - [ ] maximum likelihood
    - [ ] QFA R package for logistic model
    - [ ] Should I talk about the Development Process? (i.e. use of
      simulations and zones; determining tolerances)

** TODO Tools, solving, and fitting [1/7]
  - [ ] I wrote the code in python
  - [ ] Data analysed using Colonyzer (Python)
  - [-] Solving
    - [ ] scipy.odeint -> optimized by vectorizing and using numpy
      - [ ] atol=1.49012e-8, rtol=1.49012e-8, mxhnil=0
    - [X] libroadrunner
    - [X] must write models in SBML (libSBML) and, unlike odeint,
      simulations only produce values at constant timeintervals.
      - Cell observations from QFA are not taken at constant time
        intervals. When fitting the competition model to data,
        evaluations of the objective function require timepoints to be
        the same for observed and simulated cell density. At first, I
        dealt with this issue by simulating between adjacent
        timepoints using repeated function calls to produce simulated
        species amounts at the observed times. I used this method of
        solving to fit P15 QFA data which has cell density observation
        at 10 timepoints. A second set of QFA data from the (Stripes
        and Filled plates experiment) has \(\sim\)50 timepoints per
        plate. Solving between timepoints was therefore about five
        times slower. Instead, I made a 5th order spline (SciPy) of
        cell density timecourses and took 15 evenly spaced cell
        density datapoints from the spline. The spline used smoothing
        s=1 and timepoints ranged from time zero to the time of the
        final cell observation in the raw data.
  - [ ] SBML and reproducibility - spit out models with estimated
    parameters - mentioned in short intro.
  - [-] Fitting the Competition Model
    - [ ] SciPy's scipy.optimize.minimize (optimization and rootfinding package)
      - [ ] Constrained minimization algorithm L-BFGS-B
    - [X] Given a model and data, I made maximum likelihood estimates
      of parameters using a normal model of measurement error.
    - [X] Stopping criteria
      - Give tollerances - With these tollerances I could recover the
        parameters of simulated data sets (both with and without a
        small amount of simulated noise) with high precision.
    - [X] How did I set bounds?
      - [X] So that parameters did not hit them
      - [X] Table
    - [X] C_0 scaling
    - [ ] Two N_0 BC conditions: Cultures at the edge of a plate have an
      advantage because they have access to a greater area of
      nutrients which could diffuse from outside of the array.
      - Moved to Intro (correcting for competitoin): (Data from these
        cultures is noisey due to reflections from plate edges. This
        is only partially corrected for by Colonyzer (ref) and as a
        result data from edge cultures is useually discarded. Addniall
        et al grow the a repeats of the same culture in edge
        locations, rather than leaving them empty, because of concerns
        about competition.)
      - I correct for this (in the competition model) using a new
        parameter (\N^{e}_{t_{0}}\) for the initial amount of
        nutrients in edge cultures. In rate equations involving edge
        cultures I convert this amount to a nutrient density by
        scaling \(N_{i}\) for each culture by the ratio
        \(N^{i}_{t_{0}}/N^{e}_{t_{0}}\) where \(N^{i}_{t_{0}}\) is the
        amount of nutrients in internal cultures (which have reaction
        volume one).
      - The physical interpretation of this correction is
        that edge cultures have an extra supply of nutrients which can
        diffuse instantly into the reaction volume.
      - I improved the error in cell density estimates for cultures
        one row or column inside the edge using the method and this
        resulted in better fits to internal cultures overall.
      - I evaluated both edge cultures and internals in the objective
        function when fitting data. I selected between different fits
        to the same data the objective function for only the internal
        cultures.
      - [ ] Could not use rules to set bound NE_0 > N_0 using L-BFGS-B
        algorithm. Minor point - Haven't found a good place to
        include.
    - [X] Empties
      - Set b = 0
      - Remove from objective fn
  - [X] Fitting the logistic model
    - [X] QFA R package for logistic model
    - [X] - Contains heuristic checks for slow growing cultures and is
      currently used in QFA.
    - [X] I did not have heuristic checks but did use a plate level
      C_0. I did not have time to write heuristic checks and wanted
      to compare to existing QFA fitting.
    - [X] QFA R logistic fits do not set a plate level C_0.
  - [ ] Should I talk about the Development Process? (i.e. use of
    simulations and zones; determining tolerances) Probably won't
    have room.

** TODO Parameter Conversion [9/9]  # could go in or after fitting
  - [X] When kn = 0 reduces to logistic model with same solution
  - [X] Can equate models
  - [X] Conversion equation
  - [X] Mass action logistic assumes all nutrients go to cells and
    needs culture level N_0 to be fit.
  - [X] Can be seen as a correction to the logistic model
  - [X] Correction Figure
  - [X] We can convert between models to compare fitness estimates
    directly.
  - [X] b rank is proportional to r MDR and MDP
  - [X] b as a fitness estimate

** TODO Initail Guess [1/5]
  # I was debating putting the imaginary neighbour model in the
  # introduction or splitting the model and fitting between the
  # introduction and methods. I now think that this should all go in
  # the methods. Been as this is a masters disertation I think it
  # should be alright to have a couple of figures in the methods.
  - [X] N_0 can be inferred from final cell amounts
    - [X] Assumes reactions are nearly complete and no metabolism
    - [X] Equation for two N_0 model
  - [-] C_0 requires a rang of guesses
    - [X] logspace
    - [ ] What was the range?
  - [-] Explain the Imaginary neighbour model
    - [ ] Include a schematic
    - [X] I have come up with a model for quickly fitting individual
      cultures to get appoximate estimates of b.
    - [X] It is based on the competition model we already have.
    - [X] We fit cell data for each culture individually
      - Solve using scipy.odeint
      - Fit using scipy.optimize.minimize and least squares
    - [X] Places imaginary fast growing and empty cultures next to the
      culture of interest
    - [X] We use the same number, n, of each of these. This could be
      eight in total to represent, for instance, 4 fast growing
      neighbours or 4 slow growing neighbours.
    - [X] The value of n to use can be determined by the ratio of
      final cell amount to initial nutrient amount.
    - [X] Fix N_0 and the growth constants of both neighbours. We have
      to use a grid of fixed C_0 values and take the best fit.
    - [X] Allow k1, k2, and b to vary. b is the growth constant for
      the culture of interest.
    - [X] We can also try different values of b_fast for fits of each
      culture to approximate different neighbour configurations.
    - [X] Main difference is in having two kn's (k1 and k2) which we
      allow to vary freely.
  - [-] Guessing k_n from b distiribution
    - [X] Wordy explaination
    - [ ] I have a plot which demonstrates this which can go here or
      in the results. I think that it would be clearer to put it here.
  - [ ] Performance of imaginary neighbour guessing (plot of fit) can
    wait until the results.
** TODO Initial guess tl;dr [0/1]
  - [ ] Initial guess
    - [ ] Guessing N_0 and NE_0
    - [ ] Imaginary neighbour model - (scipy.odeint, scipy.optimize.minimize)
    - [ ] Guessing k_n from b distiributio
** TODO Development of a genetic algorithm [0/1]
  - [ ] Developing a genetic algorithm with inspyred
    - [ ] Heirarchical plate level and culture level parameters
    - [ ] Python package inspyred
      - [ ] What algorithms etc.
      - [ ] Starting values
      - [ ]
    - [ ] Multiprocessing
    - [ ] Fixing plate level and using a gradient method
      - Actually I need only mention this if I want
** TODO Model comparison using a single QFA plate [0/1]
  - [ ] P15 [0/3]
    - [ ] Describe data
    - [ ] Describe fitting and analysis
    - [ ] Describe validation spot test data here?
** TODO Cross-plate calibration and validation [0/1]
  - [ ] Stripes and Filled Plates [0/2]
    - [ ] Describe data
    - [ ] Describe fitting and analysis

- Note about availibility? This is already in the abstract

Details to add
- pandas parser
- empties
- edges
- C_0 scaling
- Bounds: Constrained minimization algorithm L-BFGS-B algorithm
  - [R128] (1, 2) Byrd, R H and P Lu and J. Nocedal. 1995. A Limited
    Memory Algorithm for Bound Constrained Optimization. SIAM Journal
    on Scientific and Statistical Computing 16 (5): 1190-1208.
  - [R129] (1, 2) Zhu, C and R H Byrd and J Nocedal. 1997. L-BFGS-B:
    Algorithm 778: L-BFGS-B, FORTRAN routines for large scale bound
    constrained optimization. ACM Transactions on Mathematical
    Software 23 (4): 550-560.
- stopping criteria
- Two N_0 model
- objective function least squares
- select just the internals
- yzer
- Splining for Stripes and Filled plates
- Can also solve SBML in Copasi

* TODO Results [0/0]
 - [ ] Try to remove mre11 repeat from data (4, 11) (with origin (0,
   0))? - Comparison of methods so could leave in. Or just check.
* TODO Discussion [0/4]
** TODO Discuss results and place in context of other work [1/4]
 - [-] P15
   - [-] Qualitatively good
     - [X] Better than logistic or generalised
     - [ ] may be overfitting?
   - [-] Fitness rankings
     - [X] Agree for top and bottom ranks
     - [X] Disagreement inbetween
     - [X] I could also do with correlation plots between model r as I
       have for the stripes and filled fits to make a comparison
     - [X] Agree with validation spot data (add refs) - detail in the
       results.
     - [ ] (Wait untill COV to discuss differences in r and MDR of
       logistic QFA.)
   - [-] COV
     - [X] COV higher in Competition model for fastest growers
     - [X] COV lower in competiton model for other cultures and slowest
       growers (dominated by noise)
     - [X] Probably due to sharing of plate level parameters, but not
       necessarily modelling of competition -> Must use data from
       noise dominated.
     - [X] QFA R reqsuires hueristic checks for slowest growers due to
       confounding between r and K.
     - [X] Still some issues - Some encroachment not dealt with picked
       by Colonyzer of QFA R
     - [X] - est1 fitted with a much higher r and and low K
     - [X] Order corrected when converting to MDR creating confusion
       about which fitness estimate to use.
     - [X] Could be corrected either experimentally, in Colonyzer, or in
       QFA R using a heuristic check for late growers.
     - [X] Competition model rank of slow growers probably more trustworthy
     - [ ] QFA R uses culture level C_0 (may or may not be more
       accurate)
   - [-] HETEROGENEITY: Herrmann and Lawless:
     - [X] Due to cell heterogeneity we believe that C_0 varys between
       cultures.
     - [X] Makes direct measurement of C_0 unreliable
     - [ ] Allowing a culture level C_0 more accurate but preferable
       if were similar
     - [X] Strategies:
       - [X] Experimental:
	 - [X] Inoculate cells taken from the exponential growth phase
           to reduce heterogeneity
	 - [X] Use higher starting density to try to average out
           heterogeneity
       - [X] Analysis:
	 - [X] Then: Fit C_0 collectively -> 1 parameter.
	 - [X] Or: Only small amount of nutrients used at start -> Can
           allow cultures to grow for a short while before starting
           observations and directly measuring C_0 -> Separate direct
           measurements of all cultures.
	 - [X] Or: Fit separately without direct measurement -> 384
           parameters
   - [X] Conclusion: Failure to improve reliability over Addinall et
     al and slower, although probably better for slow groweres and I
     can suggest to trust MDR over r due to failing heuristic checks
   - [X] Less significance in fitnesses estimates for culture ranked
     in the middle because comparison is to a neutral deletion so
     change in order not so important
 - [-] Stripes and Filled
   - [X] Starts with higher inoculum so no noise-dominated cultures.
   - [X] No heuristic checks needed for logistic when no noise dominated
   - [X] Performance of competition model improved and similar to
     logistic model. Need correlation plot for P15 to compare.
     - [X] Despite not finding global minima
   - [X] However correlation between models is now worse
   - [X] Validation issues with both models.
     - [X] Logistic does not correct at all for competition
     - [X] Competitoin model overcorrects
   - [X] No repeats on this plate so cannot study COV
   - [-] How can we determine which estimates to believe?
     - [X] Do we lack validation data for this plate?
     - [ ] Could do with new experiments - FUTURE WORK
       - [ ] Softer validation - More representative of actual
         experiment (i.e. less drastic difference).
       - [ ] Repeat of P15 with higher C_0.
       - [ ] Ideally leave edge cultures empty
       - [ ] Deliberately try to induce a change in rank by targeting
         different strains with different neighbours.
 - [X] Issues with fitting
   - [X] Not finding global minina
   - [X] Could try genetic algorithm or Bayesian approach of
     Heyderi et al.
     - [X] Bayesian likely to be slower
     - [X] I have found that fixing plate level parameters allows
       global minima to be found so could possibly use hierarchical ga
       with candidates of plate level parameters. Could be faster.
   - [X] However, fitness rankings are similar for best fits and also
     make overestimates in validation so likely to be a more
     fundamental issue with the model.
 - [-] Conclusion:
   - [ ] Competitoin model appears to perform better than in p15
     (comparable?) and has different order to logistic (poor
     correlation)
   - [ ] Probably requires futher validation using cultures with
     inoculated higher density, repeats, and known biology. Perhaps a
     repeat of P15.
   - [X] In any case the competitoin model still needs improving


** TODO Future Work [0/4]
 - [ ] Discussion about differences in final cells could potentially
   go to the introduction or methods
 - [-] Validate the independent limit
   - [-] Assume no metabolism
     - [X] mentioned but not explained next 3 points in detail
     - [ ] Could differ between strains
     - [ ] Could eat the limiting nutrient affecting reaction rate
     - [ ] Hypothetically could account entirely for difference in
       final cell density but would not cause observed interactions.
   - [X] Mass action kinetics (I have a ref for this if needed)
     - [X] more valid for liquid cultures
   - [X] Liquid cultures (paper)
   - [-] Isolated spots
     - [X] Different dimensions - Issues with diffusion limited and
       surface (Fractal kinetics may be required (ref))
     - [ ] Cultures not uniform (ring around outside)
     - [X] cultures grow differently between mediums but ok. Same
       model different parameters.
   - [ ] Dealing with fitness: Which is more like yeast's natural
     environment? Don't think this matters. Will just reveal different
     interactions/processes/circuits.
   - [-] Differences between nutrient density.
     - [ ] Logistic model -> diffusion limited inside cultures?
     - [X] I believe diffusion is limited in supply from agar
       - [X] that nutrients are not well approximated as being evenly
         distributed withing the spatial scales that we model.
     - [ ] Identify the limiting nutrient modecule
     - [ ] Design agar to limit a certain nutrient that is not
       metabolised.
   - [X] Our fits contain fairly high values of kn (fast diffusion)
     may be inaccurate
   - [X] Growth may stop before nutreints diffuse form neighbour.
     - [X] Can experimentally determine this also by growing new
       cultures at distances after first culture reaches stationary
       phase.
     - [X] Still requires simulation to study over depth.
   - [ ] In an experiment to test the length-scale of the interaction
     could inoculate a gradient of fast to slow growing cultures
     across a plateand observe local differences in endpoint.
   - [X] Diffusion equation model to study point where growth is
     limited by supply/diffusion of nutrients through agar.
     - [X] A diffusion equation model could capture the local
       distribution of nutrients around a culture when the stationary
       phase is reached
     - [X] I have a good section on Reo and Korolev that should go in
       either the introduction or here.
   - [X] May require finer grid.
   - [X] Signal could instead/also be modelled
     - [X] Can be easily done using CANS by changing the raction
       equations.
     - [X] Ethanol or quorum sensing (ref)
   - [ ] Arrest. Sick cells may be arresting
   - [X] Combination of effects hard to fit without extra observations
     - [X] May have to come up with ways to calibrate effects in
       isolation.
 - [ ] Limitation on computation time for fitting many plates in
   high-throughput restricts possible modelling apporaches.
   - [ ] How many plates?
   - [ ] How long per fit at the moment?
 - [ ] Which fitness estimates to use. Kill the generalised logistic
   model.


** TODO Improvements to current code / workflow [3/4]
  - [X] loop through b_guess in Imag neigh guessing
  - [X] Alternatively could use logistic model as init guess
  - [-] Edge cultures [1/2]
    - [ ] Discussion: An alternative model could have added nutrient
      containing volumes (essentially empty cultures) outside of edge
      cultures with an extra parameter for their size and included
      theses in the network diffusion model with the same \(kn\). As
      this is just a correction and cell density observations from these
      cultures are very noisy, and these cultures are dominated by
      noise.
    - [X] Discussion: When fitting the competition model, it would be
      better if these cultures were left empty to remove noise in cell
      density observations.
  - [X] To make development easier use smaller isolated zones inside a
    full plate so that boudaries are easier to treat.
* TODO Conclusion [0/0]
* TODO Acknowlegements [0/6]
  # How do I go about acknowledge people involved in the work? / What do I need to do?
  - [ ] Conor - Coming up with the model and idea for the project. Collaborated throughout.
    - Should I reference some of your blog posts?
  - [ ] Paolo - Suggested to use roadrunner for instance
  - [ ] Helena - I wish to talk about her project to discuss issues
    with C_0
  - [ ] Addinall - P15
  - [ ] Who? - Stripes and Filled
  - [ ] David - Validation spot test data


* TODO Intro ammendments [1/9]
 - [ ] (Proirity 1) Sharped Plate images
 - [ ] (Priority) Decide whether I like having the Reo model in the introduction.
 - [ ] Reference Fisher for genetic interaction (Not going to)
 - [ ] Could talk about more examples of QFA
   - [ ] If I am having to introduce P15 and the Stripes and Filled
     plates earlier then talk about P15
 - [-] Correct statements about yeast genome and growth [4/7]
   - [ ] Not really certain that this is why they have small
     genomes. Unless I can think of another evolutionary pressure this
     will have to do though.
   - [X] Orthologs Inparanoid O'Brien
   - [ ] Do not state as Beguling (compact cells more efficient) Doolittle
   - [X] Streamlining theory Giovannoni
   - [X] Deth 2007 - "Performance of the translational apparatus
     varies with the ecological strategies of bacteria"
   - [X] life with 6000 genes (yeast genome)
 - [ ] (Good but low priority) Plots of r and K dists for stripes and
   filled
 - [ ] Could mention quorum sensing when I menthion ethanol
 - [ ] Could put arrows, boxes, numbers, or letters on stripes and
   filled rows
 - [X] Change kn-k and C_{t_{0}}->C(0), etc. in all figures

* TODO Methods ammendments [1/6]
 - [-] Find solving and fitting time
   - [ ] Comp model
   - [ ] Logistic model (Individually (minutes) and collectively (?))
   - [X] Imag neigh guess - fitting
 - [ ] Need to explain plates earlier and edit notes and references in
   the text.
 - [ ] Schematic of initial guess
 - [ ] Derive conversion equations: Only if room.
 - [X] Remove kn from corerction plot
 - [ ] (low priority) Draw all correction cultures on one
   plot. Probably not.

* TODO Results ammendments [0/1]
 - [ ] (low priority) Could remove middle data in validation plot for
   stripes and plot the logistic fit as well.
* TODO Discussion [0/4]
 - [ ] (Priority) Compare least squares of comp and logistic model or plot the
   fits
 - [ ] (Priority) Sort validation refs and check agreement
 - [ ] p-values on figure
 - [ ] (Priority) r correlations for P15
